certora/specs/Marketplace.spec

```diff
@@ -71,11 +71,22 @@ hook Sload bool defaultValue _missing[KEY MarketplaceHarness.SlotId slotId][KEY
 
 hook Sstore _missing[KEY MarketplaceHarness.SlotId slotId][KEY Periods.Period period] bool defaultValue {
     _missingMirror[slotId][period] = defaultValue;
-    _missedCalculated[slotId] = _missedCalculated[slotId] + 1;
+    if (defaultValue) {
+        _missedCalculated[slotId] = _missedCalculated[slotId] + 1;
+    } else {
+        _missedCalculated[slotId] = _missedCalculated[slotId] - 1;
+    }
```

Comment:
So we have this `missedCalculated[slotId]` ghost, which aims to keep track of how many times a slot has been marked as missed.

We're hooking into `_missing` map `SSTORE` operations. In other words, whenever a slot period is `missing` is set `true` we increase the counter. Otherwise we decrease it.

I don't actually think there's a case currently in code where it's set `false` so it should only increase atm.

Reply:
> I don't actually think there's a case currently in code where it's set false so it should only increase atm.

Fair, I just don't understand how you want to employ the decreasing logic in some potential tests? It seems to me like a construct that could lead to some unexpected behavior or potentially hiding some bugs? I would have expected tests like "did proof got missed" or similar using assertions on this array by expecting the slot's counter to increase. If there is option that it could also decrease, then you could lend in a state which would have missed proof but it would not be detected...

Reply:
I'm going to double check this and update the hook if necessary

Reply:
Invariant passes without decrease, gonna remove that.
Here's the prover run: https://prover.certora.com/output/6199/6a076ed7e7a149abaa17d412edc97163?anonymousKey=4eb37acdfecbef4ab032f070b5bc124730d86fad

Reply:
Updated the PR.